typeform/typeform.py

```diff
@@ -0,0 +1,183 @@
+import time
+import copy
+import json
+import panoply
+import urllib2
+import datetime
+
+
+FETCH_LIMIT = 500
+DESTINATION = 'typeform'
+DESTINATION_POSTFIX = '_{__table}'
+BASE_URL = 'https://api.typeform.com/v1'
+DATE_PARSER_FORMAT = '%Y-%m-%dT%H:%M:%S'
+
+
+class Typeform(panoply.DataSource):
+
+    # constructor
+    def __init__(self, source, options):
+        super(Typeform, self).__init__(source, options)
+
+        if 'destination' not in source:
+            source['destination'] = DESTINATION
+
+        # append the destination postfix, which represent the form name
+        source['destination'] += DESTINATION_POSTFIX
+
+        # configure incremental pulling
+        self._incval = None
+        if (source.get('inckey')):
+            incval = source.get('incval')
+            self._incval = (incval if incval else
```

Comment:
Well at my first attempt on this, the user wouldn't have defined his own `inckey` - it just gives him the option to choose whether to perform incremental pulling and using the `lastTimeSucceed`/`lastRuntime` as the default option if he hasn't chosen otherwise. 
But thinking on this again, this data source _does_ support incremental pulling since we're only fetching **completed** surveys and therefore we can use the `since` variable from the last successful retrieval. thanks.